presto-cassandra/src/main/java/com/facebook/presto/cassandra/CassandraMetadata.java

```diff
@@ -113,15 +113,14 @@ private static SchemaTableName getTableName(ConnectorTableHandle tableHandle)
     public ConnectorTableMetadata getTableMetadata(ConnectorSession session, ConnectorTableHandle tableHandle)
     {
         requireNonNull(tableHandle, "tableHandle is null");
-        SchemaTableName tableName = getTableName(tableHandle);
-        return getTableMetadata(session, tableName);
+        return getTableMetadata(getTableName(tableHandle));
     }
 
-    private ConnectorTableMetadata getTableMetadata(ConnectorSession session, SchemaTableName tableName)
+    private ConnectorTableMetadata getTableMetadata(SchemaTableName tableName)
     {
-        CassandraTableHandle tableHandle = schemaProvider.getTableHandle(tableName);
-        List<ColumnMetadata> columns = getColumnHandles(session, tableHandle).values().stream()
-                .map(column -> ((CassandraColumnHandle) column).getColumnMetadata())
+        CassandraTable table = cassandraSession.getTable(tableName);
+        List<ColumnMetadata> columns = table.getColumns().stream()
```

Comment:
Columns with the same lowercase name are not supported =\
This is because we provide column handles to Presto in a map.
The check for column name compatibility is performed in NativeCassandraSession#164